### ISSUE TITLE
Consider development and publication process update

### ISSUE THREAD

**Issue by mkondratek** [VirtusLab/git-machete-intellij-plugin#261]
**develop - master** branch pair is an option:

> each PR when merged goes to develop
when develop accumulates changes (in particular just one PR) that are ready to be released publicly, then develop is merged into master (or just master is FFed to match develop);
each merge to master triggers an automatic release (in our case, plugin publish)
also: the versioning scheme would likely need to be adapted, TBD

_Originally posted by @PawelLipski  on Slack_

This will possibly resolve #193  as well

**Comment by PawelLipski** [comment]
Wrt. new versioning scheme: pre-release suffix like `-1`, `-2` etc. can be used (even without specifying `-RC`, `-M`, `-alpha`, `-beta` etc., just a number to introduce a clear ordering of the pre-release versions)

**Comment by PawelLipski** [comment]
Also, the following scenario is likely that after a release e.g. `1.0.3`, we'll have the following sequence of pre-release versions:

* `1.0.4-1`
* `1.0.4-2`
* `1.0.4-3`
* `1.1.0-1` - coz we've just adding new feature, the new release won't be a PATCH-level anymore, but MINOR-level one
* `1.1.0-2`
* `1.1.0-3`
* `1.1.0` - finally releasing as minor release; as a consequence, `1.0.4` never actually gets released

**Comment by PawelLipski** [comment]
TBD how develop->master release should be done, preferably a separate CI job that:

* can only ever be triggered manually from CI (and not automatically after anything is pushed)
* commits the pre-release suffix removal from the version on master
* pushes the updated master, hence triggering a regular master build

Each successful master build, in turn, should publish the plugin artifact to Marketplace as its final step.

**Comment by PawelLipski** [comment]
Whoops, this part:

> can only ever be triggered manually from CI

seems pretty hard to achieve on Circle CI (https://discuss.circleci.com/t/manually-triggered-deployment-stages/5515/28), the closest we can get is:

https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval

**Comment by PawelLipski** [comment]
So likely we should instead provide a script that would run the steps 1-4 locally (TBD how to handle master branch protection then?), and master builds can publish the plugin - but only after getting a *manual approval* in Circle CI.
